core/src/main/java/tech/tablesaw/api/LongColumn.java

```diff
@@ -244,10 +244,10 @@ public LongColumn append(LongColumn column2) {
         return result;
     }
 
-    public LongColumn subtract(LongColumn column2) {
+    public LongColumn subtract(NumericColumn column2) {
```

Comment:
Should a longColumn subtracting a double return a long column or a double column?

Reply:
Interesting question.
Looking at the standards, it should be double column (for e.g. `2 - 1.0 = 1.0`). However, It seems to me it will demand significant overloading. For e.g. `int - long` will be handled different from `int - double`. Or it might impact performance if additional type checks are required for every row. I am also concerned that changing the type might interfere with operator chaining.

As of now I am inclining towards returning the same type as that of the column on which the operation is invoked. What do you think?

Reply:
Or what about `2 - 0.5`? I'd expect it to return `1.5`

I'd expect that if we have have two columns being combined that we use the more precise type regardless of the order (i.e. `2.5 - 1` and `2 - .5` should both return `1.5`):
long, int -> long
long, short -> long
int, short -> int
double, anything -> double
float, anything-but-double -> float

Reply:
Sounds good to me. This approach prevents data loss. I will try and push an update. Thanks !

Reply:
Actually, `long` to `float` conversion could be concerning in some cases. For e.g. if a `LongColumn` contains `System.currentTimeMillis()`. A `float` conversion of this column can result in error of a few minutes. I am thinking of not supporting auto `long` to `float` conversion. 

```
float, anything-but-double-&-long -> float
```

Reply:
yeah, maybe if you have a long and float you should convert them both to doubles?